sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -19,32 +19,34 @@ package org.apache.spark.sql.catalyst.util
 
 import java.util.regex.{Pattern, PatternSyntaxException}
 
+import org.apache.spark.sql.AnalysisException
 import org.apache.spark.unsafe.types.UTF8String
 
 object StringUtils {
 
-  // replace the _ with .{1} exactly match 1 time of any character
-  // replace the % with .*, match 0 or more times with any character
-  def escapeLikeRegex(v: String): String = {
-    if (!v.isEmpty) {
-      "(?s)" + (' ' +: v.init).zip(v).flatMap {
-        case (prev, '\\') => ""
-        case ('\\', c) =>
-          c match {
-            case '_' => "_"
-            case '%' => "%"
-            case _ => Pattern.quote("\\" + c)
-          }
-        case (prev, c) =>
+  /** Convert 'like' pattern to Java regex. */
+  def escapeLikeRegex(str: String): String = {
+    val in = str.toIterator
+    val out = new StringBuilder()
+
+    def fail(message: String) = throw new AnalysisException(
+      s"the pattern '$str' is invalid, $message")
```

Comment:
>  org.apache.spark.sql.AnalysisException: the pattern '\a' is invalid, the escape character is not allowed to precede 'a';

or 

> org.apache.spark.sql.AnalysisException: The pattern '\a' is invalid. Reason: the escape character is not allowed to precede 'a';
 
Isn't this just a matter of preference? I preferred the first one because it avoids using sentences that are expected to end in a period, when a semi colon is always appended after an analysis exception. It's a really really superficial thing though, so I'm happy to change it either way